frontend/src/features/map_planning/hooks/useGetLayers.tsx

```diff
@@ -0,0 +1,15 @@
+import { getLayers } from '../api/getLayers';
+import { useQuery } from '@tanstack/react-query';
+
+/**
+ * Gets all layers for the given map id.
+ * Re-fetches data only after a certain amount of time to reduce network stress.
```

Comment:
Why refetch at all? Can't we send a SSE if a new layer is created? (please create issue if this is a bigger thing to do).

Reply:
Yes, this would work.
In this case, the action sent from the server could just be a notification.
Nevertheless, I would refetch the layers on receiving the notification.
Otherwise, merging the existing layers in `react-query` with the incoming layers would be complicated.

Reply:
Yes, fetching all layers again when there was a change with the layers sounds reasonable.

Reply:
I added this issue: https://github.com/ElektraInitiative/PermaplanT/issues/498